src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/CodeStyle/CodeStyleOption2`1.cs

```diff
@@ -82,7 +82,7 @@ public NotificationOption2 Notification
             new("CodeStyleOption", // Ensure that we use "CodeStyleOption" as the name for back compat.
                 new XAttribute(nameof(SerializationVersion), SerializationVersion),
                 new XAttribute("Type", GetTypeNameForSerialization()),
-                new XAttribute(nameof(Value), GetValueForSerialization()),
+                new XAttribute(nameof(Value), GetValueForSerialization()!),
```

Comment:
It's only nullable when using uninitialized value with T=string. There could be a later refactor on that.

Reply:
I guess I'd say in that case we should:

1. annotate GetValueForSerialization to return not-null.
2. In the branch for string, do a Contract.ThrowIfNull().
3. In the branch for bool, just ! since that's more obvious what's going on here.

Now I'm also a bit confused because the docs for the type say that T is only allowed to be bool or an enum, so clearly we are breaking that rule somewhere...? @mavasani or @jmarolf any idea if nulls are supposed to be supported here?

Reply:
@jmarolf confirmed that we don't expect to get nulls here, so current code is good.